### ISSUE TITLE
CodeStar1401-SE-Phase09-Team06

### ISSUE THREAD

**Issue by ali-bonaftizadeh** [Star-Academy/codestar-intern-issues#698]
## Web history

- [x] Before starting: Read the link about history and purpose of creating & developing Web, then answer this questions:
  - [x] How did developers provide server-client softwares for users before creation of web technologies?
  
      `Actually, a very important part of the idea of web technologies was basically separating client and server. But to answer this question we can say before the creation of web technologies, apps were created for different platforms and architectures making them hard to update and develop.`

  - [x] What was the main trouble with traditional technologies before web?
  
    `Apps were dependent on the client machine, its operating systems, and architecture. So, it was hard to make an app runnable on different computers with different configs. `

  - [x] How did web overcome this main trouble?
  
      `Web technology made it possible to have cross-platform apps running on different operating systems so that developers no longer needed to design a different version of an app for each OS.`

## ASP.NET Core

- [x] Getting Started with HTTP:
  - [x] What does a protocol mean?
  
    `A protocol is a system of rules that define how data is exchanged within or between computers.`

  - [x] Why designing a protocol for communication is important prerequirement of all other web related technologies?
  
    `Because the protocol determines the rules and constructions between client and server. as a result, it's the most basic requirement for web apps.`

  - [x] Try to get familier with HTTP by reading links.
  - [x] Why HTTP is most popular protocol in web technologies?

    `Because it can transfer all types of data like text, layout description, images, videos, scripts, and more, also, it's readable for humans.`

  - [x] What is the most interesting aspect of HTTP in your opinion?
  
    `It's interesting that we can return any type of object and receive that without dealing with parsing objects.`

- [x] Getting Started with REST:
  - [x] Read mentioned links.
  - [x] Why do we need RESTful APIs?
  
    `RESTful APIs help us classify requests and handle different types of requests easier and cleaner.`

- [x] Getting Started with Asp.net Core:
  - [x] Read mentioned links.
  - [x] Create a hello world API.
  - [x] Call your api from browser.
  - [x] Play with routing options! (Try to use it & enjoy it!)
  - [x] Why do we need configurable routings in web development?
  
    `To map requests to it's respected Action, we need to define routes and specify a route for every Action.`
  
  - [x] Play with dependency injection! (Define an interface & inject interface implementation to code)
  - [x] Why do we need dependency injection?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
  - [x] Play with parameter binding! (use different modes)
  - [x] Install postman.
  - [x] Call your APIs with postman.
  
- [x] Implement Asp.net Core Project:
  - [x] Design your search API.
  - [x] Implement & Test your API.

## Review

Link of your own PR:

https://github.com/Star-Academy/Summer1401-SE-Team06/pull/7

Link of other PR(s) you agreed to review:

`[FILL HERE WITH LINK OF PR(s) YOU REVIEWED.]`

- [x] Your own PR is reviewed and approved by at least one other team.

- [x] Your own PR is reviewed and approved by mentor (mentor 2).

- [x] Your own PR is reviewed and approved by manager (mentor 1).

- [x] Your own PR is merged.
